privacyidea/api/lib/prepolicy.py

```diff
@@ -991,6 +991,58 @@ def check_anonymous_user(request=None, action=None):
     return True
 
 
+def check_admin_tokenlist(request=None, action=None):
+    """
+    Depending on the policy scope=admin, action=tokenlist, the
+    filterRealms parameter is set to define, the token of which
+    realms and administrator is allowed to see.
+
+    Sets the filterRealm
+    None: means the admin has no restrictions
+    []: the admin can not see any realms
+    ["realm1", "realm2"...]: the admin can see these realms
+
+    :param request:
+    :return:
+    """
+    filterRealm = None
+    wildcard = False
+    role = g.logged_in_user.get("role")
+    if role == ROLE.USER:
```

Comment:
We can keep it like that, I just had an idea: but what do you think about restructuring the whole function like this
```python
if g.logged_in_user.get("role") != ROLE.USER:
    # everything else
return True
```
? I think this makes it clearer that it only takes effects for admins.

Reply:
Will check and decide.

Reply:
done in 80747ff6